### DISCUSSION TITLE
Button Click event calling twice

### DISCUSSION THREAD

**Discussion by vignesh-sivaprakasam** [twilio-labs/paste#2085] [General] [answered]
I have a use case where I have to wrap checkbox with button.
* when I click on button event is firing once whereas when i click on checkbox click event is fired twice.

Code sandbox: https://codesandbox.io/embed/admiring-hugle-pmzzu?fontsize=14&hidenavigation=1&theme=dark

**Comment by loreina** [comment, marked as answer]
Hi @vignesh-sivaprakasam! 

From an accessibility perspective, a checkbox should not be nested within a button as you can't nest focusable HTML elements. In the [HTML spec](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button), it highlights that permitted content in a Button includes "Phrasing content but there must be no Interactive content". 

Could you share more about your use case? Happy to discuss possible accessible options to the interaction you're looking for.

**Comment by andioneto** [comment]
Hi @vignesh-sivaprakasam 👋 Do you have what you need or would you like more assistance with this?
